File: maxProd.py
def max_product(nums):
    if not nums:
        return 0

    n = len(nums)
    max_product = nums[0]
    min_product = nums[0]
    result = nums[0]

    for i in range(1, n):
        temp = max_product
        max_product = max(nums[i], max_product * nums[i], min_product * nums[i])
        min_product = min(nums[i], temp * nums[i], min_product * nums[i])
        result = max(result, max_product)

    return result

if __name__ == "__main__":
    n = int(input())
    nums = list(map(int, input().split()))

    max_product_value = max_product(nums)
    print("Maximum Product Subarray:", max_product_value)

File: maxProd.java
import java.util.Scanner;

public class MaxProductSubarray {
    public static int maxProduct(int[] nums) {
        int n = nums.length;
        if (n == 0) {
            return 0;
        }

        int maxProduct = nums[0];
        int minProduct = nums[0];
        int result = nums[0];

        for (int i = 1; i < n; i++) {
            int temp = maxProduct;
            maxProduct = Math.max(nums[i], Math.max(maxProduct * nums[i], minProduct * nums[i]));
            minProduct = Math.min(nums[i], Math.min(temp * nums[i], minProduct * nums[i]));
            result = Math.max(result, maxProduct);
        }

        return result;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] nums = new int[n];
        for (int i = 0; i < n; i++) {
            nums[i] = scanner.nextInt();
        }
        scanner.close();

        int maxProduct = maxProduct(nums);
        System.out.println("Maximum Product Subarray: " + maxProduct);
    }
}
